addons/sourcemod/scripting/shavit-tas.sp

```diff
@@ -0,0 +1,1395 @@
+{
+	if(!gB_TAS[client])
+	{
+		return Plugin_Continue;
+	}
+	gI_Status[client] = PAUSED;
+	return Plugin_Handled;
+}
+
+public void OnClientPutInServer(int client)
+{
+	
+	if(gA_Frames[client] != INVALID_HANDLE)
+	{
+		ClearArray(gA_Frames[client]);
+	}
+	else
+	{
+		gA_Frames[client] = CreateArray(11, 0);
+	}
+
+	gF_CounterSpeed[client] = 1.0;
+	gF_TASTime[client] = 0.0;
+	gF_Timescale[client] = 1.0;
+	gI_Status[client] = RUN;
+	gB_TASMenu[client] = true;
+	gB_AutoStrafeEnabled[client] = false;
+	gB_SilentStrafe[client] = false;
+	gI_Type[client] = Type_SurfOverride;
+	gF_Power[client] = 1.0;
+}
+
+public Action OnPlayerRunCmd(int client, int &buttons, int &impulse, float vel[3], float angles[3], int &weapon, int &subtype, int &cmdnum, int &tickcount, int &seed, int mouse[2])
+{
+	if (IsValidClient(client, true) && gB_TAS[client])
+	{
+
+		if(!gB_TAS[client])
+		{
+			return Plugin_Continue;
+		}
+
+		DrawPanel(client);
+
+		if(Shavit_GetTimerStatus(client) != Timer_Running)
+		{
+			return Plugin_Continue;
+		}
+		else if(IsPlayerAlive(client) && !IsFakeClient(client))
+		{
+			if(gI_Status[client] == RUN)
+			{ // Record Frames
+				//SetEntPropFloat(client, Prop_Send, "m_flLaggedMovementValue", gF_Timescale[client]);
+				float fTimeScale = GetEntPropFloat(client, Prop_Data, "m_flLaggedMovementValue");
+				gF_TimescaleTicksPassed[client] += fTimeScale;
+
+				float fDifference = angles[1] - gF_LastAngle[client];
+				if (fDifference > 180.0)
+				{
+					fDifference -= 360.0;
+				}
+				else if(fDifference < -180.0)
+				{
+					fDifference += 360.0;
+				}
+				/*
+							AUTO STRAFER START
+														*/
+				if(buttons & IN_FORWARD && vel[0] <= 50.0)
+				{
+					vel[0] = 450.0;
+				}
+
+				float yaw_change = 0.0;
+				if(vel[0] > 50.0)
+				{
+					yaw_change = 30.0 * FloatAbs(30.0 / vel[0]);
+				}
+
+				if (gB_AutoStrafeEnabled[client] == true && Shavit_GetTimerStatus(client) == Timer_Running && gB_TAS[client] && !(GetEntityFlags(client) & FL_ONGROUND) && (GetEntityMoveType(client) != MOVETYPE_NOCLIP) && !(buttons & IN_FORWARD) && !(buttons & IN_BACK) && !(buttons & IN_MOVELEFT) && !(buttons & IN_MOVERIGHT))
+				{
+					if(fDifference < 0.0)
+					{
+						angles[1] += yaw_change;
+						//buttons |= IN_MOVERIGHT;
+						vel[1] = gF_MaxMove;
+					}
+					else if(fDifference > 0.0)
+					{
+						angles[1] -= yaw_change;
+						//buttons |= IN_MOVELEFT;
+						vel[1] = gF_MaxMove * -1.0;
+					}
+				}
+				/*
+							AUTO STRAFER END
+														*/
+
+				/*
+						WIGGLEHACK START
+						Huge amounts of credit to Kamay for this code!
+						https://steamcommunity.com/id/xutaxkamay/
+																		*/
+				if (gB_SilentStrafe[client] == true && Shavit_GetTimerStatus(client) == Timer_Running && gB_TAS[client] && !(GetEntityFlags(client) & FL_ONGROUND) && (GetEntityMoveType(client) != MOVETYPE_NOCLIP) && !(buttons & IN_FORWARD) && !(buttons & IN_BACK) && !(buttons & IN_MOVELEFT) && !(buttons & IN_MOVERIGHT))
```

Comment:
remove explicit true, remove implicit bool, and break into more lines, and timer status again.